desc/utils.py

```diff
@@ -608,3 +608,71 @@ def unique_list(thelist):
 def is_any_instance(things, cls):
     """Check if any of things is an instance of cls."""
     return any([isinstance(t, cls) for t in things])
+
+
+def broadcast_tree(tree_in, tree_out, dtype=int):
```

Comment:
Unfortunately not, because `tree_map` requires both trees to have the same structure. In our case we want to allow `tree_in` to be a "subset" of `tree_out`, which then gets broadcast to the proper structure.